Resources/NetHook2/NetHook2/crypto.cpp

```diff
@@ -53,33 +41,34 @@ CCrypto::CCrypto()
 
 	g_pLogger->LogConsole( "CCrypto::SymmetricEncryptChosenIV = 0x%x\n", Encrypt_Orig );
 
-	char *pGetMessageList = NULL;
-	bool bGetMessageList = steamClientScan.FindFunction(
+	char *pGetMessageList = nullptr;
+	const bool bGetMessageList = steamClientScan.FindFunction(
 		"\xA1\x00\x00\x00\x00\xA8\x01\x75\x29\x68\x00\x00\x00\x00\x83\xC8\x01\xB9\x00\x00\x00\x00\x68\x00\x00\x00\x00\xA3\x00\x00\x00\x00\xE8\x00\x00\x00\x00\x68\x00\x00\x00\x00\xE8\x00\x00\x00\x00\x83\xC4\x04\xB8\x00\x00\x00\x00\xC3",
 		"x????xxxxx????xxxx????x????x????x????x????x????xxxx????x",
 		(void **)&pGetMessageList
 	);
 
 	if (bGetMessageList)
 	{
-		const uint32 uMessageListStartPtrOffset = 23;
-		const uint32 uMessageListCountPtrOffset = 10;
+		constexpr uint32 uMessageListStartPtrOffset = 23;
+		constexpr uint32 uMessageListCountPtrOffset = 10;
 
 		MsgInfo_t *pInfos = *(MsgInfo_t **)( pGetMessageList + uMessageListStartPtrOffset );
 		const uint32 uNumMessages = *(uint32 *)( pGetMessageList + uMessageListCountPtrOffset );
 
 		g_pLogger->LogConsole( "pGetMessageList = 0x%x\npInfos = 0x%x\nnumMessages = %d\n", pGetMessageList, pInfos, uNumMessages );
 		
 		HMODULE sc = GetModuleHandle("steamclient.dll");
+		assert(sc != nullptr);

```

Comment:
Should this fail with an error message instead? Someone could be loading into a process that hasn't loaded steamclient yet.

Reply:
Our own injector checks for it pre-injection and has error handling for this already - see `k_ESteamProcessSearchErrorTargetProcessDoesNotHaveSteamClientDllLoaded`.